### PR TITLE
Added cookbookbat foods to diet

### PR DESCRIPTION
**Pull request by trewqpoiuuy**

### CODE REVIEW

File: src/tasks/diet.ts [hunk 1]
```diff
@@ -260,6 +281,17 @@ function menu(): MenuItem<MenuData>[] {
...
+    new MenuItem($item`Pete's rich ricotta`, { priceOverride: mallPrice($item`St. Sneaky Pete's Whey`) * 2 }),
+    new MenuItem($item`roasted vegetable focaccia`, { priceOverride: mallPrice($item`Yeast of Boris`) * 2 +  mallPrice($item`Vegetable of Jarlsberg`) * 2 }),
+    new MenuItem($item`baked veggie ricotta casserole`, { priceOverride: mallPrice($item`St. Sneaky Pete's Whey`) * 2 +  mallPrice($item`Vegetable of Jarlsberg`) * 2 }),
+    //Plain calzone gives a +ML buff, which can potentially cause problems. Excluding it to be safe
+    //new MenuItem($item`pain calzone`, { priceOverride: mallPrice($item`St. Sneaky Pete's Whey`) * 2 +  mallPrice($item`Yeast of Boris`) * 2 }),
+    //Save Deep Dish of Legend for garbo to get that sweet, sweet familiar weight buff
+    //new MenuItem($item`Deep Dish of Legend`, { priceOverride: mallPrice($item`Yeast of Boris`) * 2 +  mallPrice($item`Vegetable of Jarlsberg`) * 2 + mallPrice($item`St. Sneaky Pete's Whey`) * 2, maximum: 1 }),
+    new MenuItem($item`Calzone of Legend`, { priceOverride: mallPrice($item`Yeast of Boris`) * 2 +  mallPrice($item`Vegetable of Jarlsberg`) * 2 + mallPrice($item`St. Sneaky Pete's Whey`) * 2, maximum: 1 }),
```
Comment (Kasekopf) [inline src/tasks/diet.ts]: We should only add the calzone and pizza of legend as possible options if we have not already consumed one this ascension, according to the mafia records.

We also need to check if we have the recipes learned for each food (and each component of the food, for the big ones).

Comment (trewqpoiuuy) [reply src/tasks/diet.ts]: Added these checks. I'll do some more testing after rollover and mark this ready for review once I'm sure everything still works correctly

**Comment by trewqpoiuuy** [comment]
Dieting seems to be working fine with the new checks, so this should be ready for review

**Comment by Kasekopf** [comment]
Looks good, thanks!

**Pull request merged**